### PR TITLE
look into local image not used when enabling a local service.

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit <SvenDowideit@home.org.au>

- [x] failing test exactly as albertdb reported in #1899
- [x] see if a fix can be made
- [ ] find out if @joshwget thinks this is a really bad idea :)

for #1899

### CODE REVIEW

**Comment by SvenDowideit** [comment]
@zimme yeah, what a convoluted change.

**Comment by joshwget** [comment]
I probably made the `StageServices` function more complicated than it really needed to be. It might be easier to reason with it if you just rewrite it to avoid some libcompose stuff (making a project just to pull images is pretty odd).

If this is used only for `ros service enable` then I think this approach will work.

File: compose/project.go [hunk 1]
```diff
@@ -245,13 +247,40 @@ func StageServices(cfg *config.CloudConfig, services ...string) error {
...
 
 	// Reduce service configurations to just image and labels
+	needToPull := false
 	for _, serviceName := range p.ServiceConfigs.Keys() {
 		serviceConfig, _ := p.ServiceConfigs.Get(serviceName)
+
+		// test to see if we need to Pull
+		var client dockerClient.APIClient
+		if serviceConfig.Labels[config.ScopeLabel] != config.System {
+			client, err = rosDocker.NewDefaultClient()
```
Comment (joshwget) [inline compose/project.go]: Probably better to initialize these before the for loop.

**Comment by SvenDowideit** [comment]
yeah - it seems complicated to me too - but it will mean that the yaml is valid, and all the other t's and i's are dotted and crossed.

I'm going to go with it for now - a bunch of this kind of thing will get changed in the move to containerd

**Pull request merged**